cmd/cert-checker/main_test.go

```diff
@@ -186,3 +191,37 @@ func TestGetAndProcessCerts(t *testing.T) {
 	test.AssertEquals(t, checker.issuedReport.BadCerts, int64(5))
 	test.AssertEquals(t, len(checker.issuedReport.Entries), 5)
 }
+
+func TestSaveReport(t *testing.T) {
+	r := report{
+		begin:     time.Time{},
+		end:       time.Time{},
+		GoodCerts: 2,
+		BadCerts:  1,
+		Entries: map[string]reportEntry{
+			"020000000000004b475da49b91da5c17": reportEntry{
+				Valid: true,
+			},
+			"020000000000004d1613e581432cba7e": reportEntry{
+				Valid: true,
+			},
+			"020000000000004e402bc21035c6634a": reportEntry{
+				Valid:    false,
+				Problems: []string{"None really..."},
+			},
+		},
+	}
+
+	tmpRoot := os.TempDir()
+	test.Assert(t, tmpRoot != "", "Couldn't find the system temporary directory")
+	tmpDir, err := ioutil.TempDir(tmpRoot, "cert-checker")
```

Comment:
You can swap in "" for tmpRoot. TempDir will assume os.TempDir if the first argument is the empty string.